src/metabase/upload/parsing.clj

```diff
@@ -29,19 +31,56 @@
     :else                                (throw (IllegalArgumentException.
                                                  (tru "''{0}'' is not a recognizable boolean" s)))))
 
-(defn- parse-date
-  "Parses a date.
+(def local-date-patterns
+  "patterns used to generate the local date formatter. Excludes ISO_LOCAL_DATE (uuuu-MM-dd) because there's
+  already a built-in DateTimeFormatter for that: [[DateTimeFormatter/ISO_LOCAL_DATE]]"
+  ;; uuuu is like yyyy but is required for strict parsing and also supports negative years for BC dates
+  ;; see https://stackoverflow.com/questions/41103603/issue-with-datetimeparseexception-when-using-strict-resolver-style
+  ;; uuuu is faster than using yyyy and setting a default era
+  ["MMM dd uuuu"        ; Jan 30 2000
+   "MMM dd, uuuu"       ; Jan 30, 2000
+   "dd MMM uuuu"        ; 30 Jan 2000
+   "dd MMM, uuuu"       ; 30 Jan, 2000
+   "MMMM d uuuu"        ; January 30 2000
+   "MMMM d, uuuu"       ; January 30, 2000
+   "d MMMM uuuu"        ; 30 January 2000
+   "d MMMM, uuuu"       ; 30 January, 2000
+   "EEEE, MMMM d uuuu"  ; Sunday, January 30 2000
+   "EEEE, MMMM d, uuuu" ; Sunday, January 30, 2000
+   ])
+
+(def local-date-formatter
+  "DateTimeFormatter that runs through a set of patterns to parse a variety of local date formats."
+  (let [builder (-> (DateTimeFormatterBuilder.)
+                    (.parseCaseInsensitive))]
+    (doseq [pattern local-date-patterns]
+      (.appendOptional builder (DateTimeFormatter/ofPattern pattern)))
+    (-> builder
+        (.appendOptional DateTimeFormatter/ISO_LOCAL_DATE)
+        (.toFormatter)
+        (.withResolverStyle ResolverStyle/STRICT))))
+
+(defn parse-local-date
+  "Parses a local date string.
 
   Supported formats:
-    - yyyy-MM-dd"
+    - yyyy-MM-dd
+    - MMM dd yyyy
+    - MMM dd, yyyy
+    - dd MMM yyyy
+    - dd MMM, yyyy
+    - MMMM d yyyy
+    - MMMM d, yyyy
+    - d MMMM yyyy
+    - d MMMM, yyyy"
```

Comment:
I decided to keep this docstring using `yyyy` instead of `uuuu` because `uuuu` is less well known and only used because we want to have strict parsing. It feels more of an implementation detail

Reply:
I have somehow never come across the `u` family of patterns before.

The following advice (from [this SO](https://stackoverflow.com/questions/41177442/uuuu-versus-yyyy-in-datetimeformatter-formatting-pattern-codes-in-java)) makes some sense:

> If you use y you should also use G. Since G is rarely used, the correct year symbol is u, not y, otherwise a non-positive year will show incorrectly.

That said, I think this is one of those "saved by the mission critical legacy code out there" situations - in practice G can only get bumped in a world where all existing software has become obsolete 🙂

Reply:
Oh nevermind, I see you're only using `y` for the docstring - a nice solution! ➕